### PR TITLE
Remove mention of confusing dummy line number on non-existing file for coqdep

### PR DESCRIPTION
**Pull request by herbelin**
**Kind:** User messages

PR #14024 reported an error on non-existent files rather than ignoring them in coqdep. This error mentioned an artificial line number which is confusing when referring to a Coq project file. This commit removes the line number.

~~On non-existent files in the Coq Project, the commit chooses to raise a warning rather than an error. This is a compromise but maybe a disputable one.~~

Summary:
- File in command line:
  - ignored before 8.14
  - 8.14 (PR #14024):
    `File "foo", line 1: No such file or directory`
  - this commit:
    ~~`Error: foo: No such file or directory.`~~ (changed) `foo.v: No such file or directory`

- File in a _CoqProject:
  - ignored before 8.14
  - 8.14 (PR #14024):
    `File "foo", line 1: No such file or directory`
  - this commit:
    `In file "bar": foo.v: No such file or directory`

### CODE REVIEW

**Comment by gares** [comment]
> On non-existent files in the Coq Project, the commit chooses to raise a warning rather than an error. This is a compromise but maybe a disputable one.

Why do you need to "compromise"? Can't you just print a better error message?

**Comment by herbelin** [comment]
> > On non-existent files in the Coq Project, the commit chooses to raise a warning rather than an error. This is a compromise but maybe a disputable one.

> Why do you need to "compromise"? Can't you just print a better error message?

I don't remember. Keeping an error would indeed be as good.

**Comment by herbelin** [comment]
I changed the first error to `foo.v: No such file or directory` and the warning to an error `In file "bar": foo.v: No such file or directory`.

One day we should adopt a convention about error messages of such tools (such as printing `*** Error`, or `Error`, or the name of the command, etc.).

File: tools/coqdep_common.ml [hunk 1]
```diff
@@ -498,7 +502,7 @@ let rec treat_file old_dirname old_name =
   let complete_name = file_name name dirname in
   let stat_res =
     try stat complete_name
-    with Unix_error(error, _, _) -> error_cannot_stat complete_name error
+    with Unix_error(error, _, _) -> raise (Cannot_stat_file (complete_name, error))
```
Comment (SkySkimmer) [inline tools/coqdep_common.ml:505]: Why this change?

Comment (herbelin) [reply tools/coqdep_common.ml:505]: So as to have different messages depending on whether the file is on the command line or in a project file (see below `treat_file_command_line` and `treat_file_coq_project`).

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
@coqbot: merge now
